analyzer/src/main/kotlin/PackageCurationProvider.kt

```diff
@@ -83,10 +82,8 @@ interface PackageCurationProviderFactory<CONFIG> : ConfigurablePluginFactory<Pac
  */
 fun interface PackageCurationProvider {
     /**
-     * Return all available [PackageCuration]s for the provided [packages], associated by the package's [Identifier].
-     * Each list of curations must be non-empty; if no curation is available for a package, the returned map must not
-     * contain a key for that package's identifier at all.
+     * Return all available [PackageCuration]s which are applicable to any of the given [packages].
```

Comment:
In the context of you commit message, we should probably make more clear here that "applicable" does *not* refer to `PackageCuration.isApplicable()`. Yes, the providers for ORT's own curation format make that call, but we don't enforce other implementers to make that call: They could return any curations that would not pass `PackageCuration.isApplicable()`. Which actually is a short-coming of our API that we should think about, IMO.

Reply:
So actually, I'm currently not seeing how this is true:

> In particular, make it impossible that providers return a matching inconsistent with `PackageCuration.isApplicable()`.

Reply:
> So actually, I'm currently not seeing how this is true:

I meant that things can only be inconsistent if they amount to at least 2 things. 
One thing cannot be inconsistent as there is nothing it can be compared with.

Before this change: A curation matches if either

1. The curation is associated with the package id in the returned `Map`
2. or the `isApplicable()` returns true for a particular package

So, 1 and 2 could be inconsistent.
I removed 1, so we have a single source of truth: `isApplicable()`.

Reply:
> Yes, the providers for ORT's own curation format make that call, but we don't enforce other implementers to make that call

I don't see how this can be true. Because with the decision we made together to separate `PackageCuration`s from `Package`s in `OrtResult` implies that matching must rely on `isApplicable`, because otherwise the association was lost.

Reply:
I'm talking about the curations returned directly from `getCurationsFor()`. There is no guarantee that these pass `isApplicable()` for all provider implementations. But ideally, that's something we should do, as it would safe us the need to check for `isApplicable()` when writing curations to the ORT result.